src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -0,0 +1,265 @@
+                });
+              return unkown_broker_id_it == partition.replicas->end();
+          });
+
+        topic_response.partitions = std::move(invalid_partitions);
+        resp_it = std::move(topic_response);
+
+        // If there are no valid partitions ...
+        if (valid_partitions_end == topic.partitions.begin()) {
+            // Then remake the partitions vector as empty
+            topic.partitions = std::vector<reassignable_partition>{};
+        } else {
+            // Otherwise remake the partitions vector to only have valid
+            // reassignable_partitions.
+            topic.partitions = std::vector<reassignable_partition>{
+              topic.partitions.begin(), valid_partitions_end};
+        }
+    }
+}
+
+template<>
+ss::future<response_ptr> alter_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    alter_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    alter_partition_reassignments_response resp;
+
+    if (!ctx.authorized(
+          security::acl_operation::alter, security::default_cluster_name)) {
+        resp.data.error_code = error_code::cluster_authorization_failed;
+        resp.data.error_message = ss::sstring{
+          error_code_to_str(error_code::cluster_authorization_failed)};
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    // TODO(@NyaliaLui) - Check if we are the controller
+
+    resp.data.responses.reserve(request.data.topics.size());
+    validate_topics(
+      request.data.topics,
+      std::back_inserter(resp.data.responses),
+      ctx.metadata_cache().node_ids());
+
+    for (const reassignable_topic& topic : request.data.topics) {
+        reassignable_topic_response topic_response{.name = topic.name};
+
+        for (const reassignable_partition& partition : topic.partitions) {
+            model::ntp ntp{
+              model::kafka_namespace, topic.name, partition.partition_index};
+
+            if (partition.replicas.has_value()) {
+                std::vector<model::broker_shard> new_replicas;
+                auto alloc_units
+                  = co_await ctx.topics_frontend().generate_reassignments(
+                    partition.partition_index, topic.name, *partition.replicas);
+                // TODO(@NyaliaLui) - If alloc_units has no value, should we
+                // return an error? Kafka does not have this notion so this
+                // would be a place that Redpanda differs
+                if (alloc_units.has_value()) {
+                    auto alloc_units_ptr = std::move(alloc_units.value());
+                    auto assignments = alloc_units_ptr->get_assignments();
+                    // TODO(@NyaliaLui) - If there is no current assignments or
+                    // the pid is incorrect, should we return an error? Kafka
```

Comment:
I see no error is returned but more importantly the code flow continues as normal to call move_partition_replicas, whats the implication of this if no units to allocate was obtianed?

Reply:
> I see no error is returned but more importantly the code flow continues as normal to call move_partition_replicas, whats the implication of this if no units to allocate was obtianed?

That's what I was trying to say in my comment(s), but I was not descriptive enough. I'm open to ideas but I'm thinking we should return some sort of "internal error" to the user.